src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ICustomNativeTypeMarshallingStrategy.cs

```diff
@@ -1003,9 +1103,99 @@ public IEnumerable<StatementSyntax> GenerateSetupStatements(TypePositionInfo inf
             return _innerMarshaller.GenerateSetupStatements(info, context);
         }
 
+        private StatementSyntax GenerateByValueUnmarshalStatement(TypePositionInfo info, StubCodeContext context)
+        {
+            // Use ManagedSource and NativeDestination spans for by-value marshalling since we're just marshalling back the contents,
+            // not the array itself.
+            // This code is ugly since we're now enforcing readonly safety with ReadOnlySpan for all other scenarios,
+            // but this is an uncommon case so we don't want to design the API around enabling just it.
+            var (_, nativeIdentifier) = context.GetIdentifiers(info);
```

Comment:
Can you also include the code being generated in comments? I do find it helpful for quickly grasping what the giant block of syntax generation should be doing.